server/src/main/java/org/opensearch/index/query/TermsQueryBuilder.java

```diff
@@ -529,6 +535,20 @@ public String getWriteableName() {
 
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
+        if (termsLookup != null && termsLookup.query() != null) {
+            QueryBuilder rewrittenQuery = termsLookup.query().rewrite(context);
+
+            SearchResponse response = context.getClient()
```

Comment:
Can we store this value and reuse it for rewrite phases?
Also can we avoid blocking calls on the search thread?

Reply:
Yeah, this will need to be asynchronous, probably similar to how the current terms lookup query is implemented:

https://github.com/opensearch-project/OpenSearch/blob/52b8b05163e8e0e265dda021a2759fd72f098047/server/src/main/java/org/opensearch/index/query/TermsQueryBuilder.java#L609-L612